src/main/java/org/logstash/plugins/inputs/http/MessageProcessor.java

```diff
@@ -0,0 +1,51 @@
+package org.logstash.plugins.inputs.http;
+
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.handler.codec.http.DefaultHttpResponse;
+import io.netty.handler.codec.http.FullHttpRequest;
+import io.netty.handler.codec.http.FullHttpResponse;
+import io.netty.handler.codec.http.HttpHeaderNames;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import io.netty.handler.codec.http.HttpVersion;
+import org.logstash.plugins.inputs.http.util.RejectableRunnable;
+
+public class MessageProcessor implements RejectableRunnable {
+    private final ChannelHandlerContext ctx;
+    private final FullHttpRequest req;
+    private final String remoteAddress;
+    private final IMessageHandler messageHandler;
+
+    public MessageProcessor(ChannelHandlerContext ctx, FullHttpRequest req, String remoteAddress,
+                            IMessageHandler messageHandler) {
+        this.ctx = ctx;
+        this.req = req;
+        this.remoteAddress = remoteAddress;
+        this.messageHandler = messageHandler;
+    }
+
+    public void onRejection() {
+        /*
+           TODO Should SO_LINGER be used here to force the termination immediately?
+           TODO This should only happen in the event of backpressure which means there's
+           TODO a potential scenario of many connection retries and fd exhaustion
+         */
+        try {
+            final DefaultHttpResponse response = new DefaultHttpResponse(HttpVersion.HTTP_1_1, HttpResponseStatus.TOO_MANY_REQUESTS);
+            response.headers().set(HttpHeaderNames.CONTENT_LENGTH, 0);
+            ctx.writeAndFlush(response);
+        } finally {
```

Comment:
Can log a throttled message here ?
for example:
```
                long now = System.currentTimeMillis();
                if (now - lastRejectMessage > 60000) {
                    logger.warn("uh oh back pressure! ");
                    lastRejectMessage = now;
                }
```

Reply:
I'm skeptic in adding this as it can alarm users for something that may happen rather frequently if traffic that is very..bursty. Also, we'd need an equivalent "ok no longer in back pressure", but I don't think it's easy to detect without keeping even more state.